### PR TITLE
Distinguish between Maven and Gradle for hint-text

### LINKED ISSUE #4778: gradle listExtensions displays mvn instructions
**Describe the bug**
In a project generated with Gradle, run `gradle listExtensions`. You should see in the end the following message: 

```
To get more information, append -Dquarkus.extension.format=full to your command line.

Add an extension to your project by adding the dependency to your project or use `mvn quarkus:add-extension -Dextensions="artifactId"`
```

**Expected behavior**
The message should be: 
```
To get more information, append <something> to your command line.

Add an extension to your project by adding the dependency to your project or use `gradle addExtension --extensions="artifactId"`
```

**Actual behavior**
It is not

**To Reproduce**
1. 
```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create -DprojectGroupId=org.acme -DprojectArtifactId=foo -DclassName="org.acme.quickstart.GreetingResource" -Dpath="/hello" -Dextensions="resteasy" -DbuildTool=gradle
```
2. `cd foo & ./gradlew listExtensions`

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Linux ggastald-laptop 5.3.6-200.fc30.x86_64 #1 SMP Mon Oct 14 13:11:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b03)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b03, mixed mode)
```
 - GraalVM version (if different from Java):  unused
 - Quarkus version or git rev: master

**Additional context**

The stacktrace shown in the beginning is already covered in #4705

### PR DESCRIPTION
**Pull request by 38leinaD**
Fixes #4778

### CODE REVIEW

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java [hunk 1]
```diff
@@ -66,8 +67,14 @@ public void listExtensions(boolean all, String format, String search) throws IOE
...
 
-            System.out.println("\nAdd an extension to your project by adding the dependency to your " +
-                    "project or use `mvn quarkus:add-extension -Dextensions=\"artifactId\"`");
+            if (this.buildFile != null && this.buildFile instanceof GradleBuildFile) {
+            	System.out.println("\nAdd an extension to your project by adding the dependency to your " +
+                        "build.gradle or use `gradle addExtension --extensions=\"artifactId\"`");
+            }
+            else {
+            	System.out.println("\nAdd an extension to your project by adding the dependency to your " +
+                        "pom.xml or use `mvn quarkus:add-extension -Dextensions=\"artifactId\"`");
```
Comment (machi1990) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]:
```suggestion
                        "pom.xml or use `./mvnw quarkus:add-extension -Dextensions=\"artifactId\"`");
```

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java [hunk 2]
```diff
@@ -66,8 +67,14 @@ public void listExtensions(boolean all, String format, String search) throws IOE
                 System.out.println("\nTo get more information, append -Dquarkus.extension.format=full to your command line.");
             }
 
-            System.out.println("\nAdd an extension to your project by adding the dependency to your " +
-                    "project or use `mvn quarkus:add-extension -Dextensions=\"artifactId\"`");
+            if (this.buildFile != null && this.buildFile instanceof GradleBuildFile) {
+            	System.out.println("\nAdd an extension to your project by adding the dependency to your " +
+                        "build.gradle or use `gradle addExtension --extensions=\"artifactId\"`");
```
Comment (machi1990) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]:
```suggestion
                        "build.gradle or use `./gradlew addExtension --extensions=\"artifactId\"`");
```

**Review by machi1990 — CHANGES_REQUESTED** [review]
I suggested we use mvn and gradle wrappers instead. Otherwise, LGTM. Thanks.

**Comment by machi1990** [comment]
CI failure looks unrelated, can you try to rebase to latest master and see if that fixes the issue.

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java (hunk 1 above)
Comment (38leinaD) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]: agree

**Comment by 38leinaD** [comment]
rebased to latest master and integrated your suggested changes

**Comment by machi1990** [comment]
> rebased to latest master and integrated your suggested changes

Cool, let's see if CI is happy this time. 

Also, sorry to be nitpicking, can we also squash everything the two commits into one. 
Thanks.

**Comment by 38leinaD** [comment]
Done!
btw: that's not nitpicky. its my first contribution and I am very happy about any guidance how to contribute correctly. :-)

**Comment by geoand** [comment]
Can you please rebase on master to pick up a CI fix?

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java [hunk 3]
```diff
@@ -66,8 +67,14 @@ public void listExtensions(boolean all, String format, String search) throws IOE
                 System.out.println("\nTo get more information, append -Dquarkus.extension.format=full to your command line.");
             }
 
-            System.out.println("\nAdd an extension to your project by adding the dependency to your " +
-                    "project or use `mvn quarkus:add-extension -Dextensions=\"artifactId\"`");
+            if (this.buildFile != null && this.buildFile instanceof GradleBuildFile) {
```
Comment (gastaldi) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]: No need for null check here

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java [hunk 4]
```diff
@@ -66,8 +67,14 @@ public void listExtensions(boolean all, String format, String search) throws IOE
                 System.out.println("\nTo get more information, append -Dquarkus.extension.format=full to your command line.");
```
Comment (gastaldi) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]: I may be wrong, but I think this instruction doesn’t really work in Gradle. Can you verify this?

Comment (38leinaD) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensions.java]: you are right; i missed that.

**Review by gastaldi — APPROVED** [review]
LGTM

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**